crates/miden-objects/src/block/proven_block.rs

```diff
@@ -30,7 +30,7 @@ impl ProvenBlock {
     ///
     /// This constructor does not do any validation, so passing incorrect values may lead to later
     /// panics.
-    pub fn new_unchecked(header: BlockHeader, body: BlockBody, proof: BlockProof) -> Self {
+    pub fn new_unchecked(header: UnsignedBlockHeader, body: BlockBody, proof: BlockProof) -> Self {
```

Comment:
I haven't followed the block header discussion too closely, but should this eventually be a `BlockHeader` and if so, should it already be one now or does that come later? I.e. my understanding of the flow is proposed block -> signed -> proven. So by the time we construct the `ProvenBlock`, there should be a signed header I assume.